src/iotjs_binding_helper.c

```diff
@@ -142,12 +142,34 @@ int iotjs_process_exitcode() {
 
   jerry_value_t jexitcode =
       iotjs_jval_get_property(process, IOTJS_MAGIC_STRING_EXITCODE);
-  IOTJS_ASSERT(jerry_value_is_number(jexitcode));
+  jerry_type_t val_type = jerry_value_get_type(jexitcode);
```

Comment:
There is no need for this type of checking. Only call the `jerry_value_to_number` and extract the value. It has a quick path for numbers which does the same as you are doing here.
Pseudo code:
```
num_val = jerry_value_to_number(jexitcode)
if (have_error(num_val) {
  exitcode = 1;
  remove_error_flag(num_val);
} else {
  exitcode = iotjs_jval_as_number(num_val);
}
jerry_release_value(num_val);
```

Reply:
You're right. commit updated again.
Thanks